### PR TITLE
Ref #17727 - DatabaseName and TableName should be used for $db and $table

### LINKED ISSUE #17727: `DatabaseName` and `TableName` should be used for `$db` and `$table`
The `PhpMyAdmin\Dbal\DatabaseName` and `PhpMyAdmin\Dbal\TableName` classes should be used as a type for the database and table names instead of using plain strings.

```diff
- public function foo(string $db, string $table)
+ public function foo(DatabaseName $db, TableName $table)
```

### PR DESCRIPTION
**Pull request by Arpit-Gulgulia**
Signed-off-by: Arpit-Gulgulia <arpitgulgulia@gmail.com>

### Description

Please describe your pull request: Fix for 17727

Ref #17727

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by williamdes** [comment]
Please try to put some attention to your PR titles ;)
It helps while reading mails or PR lists

**Comment by Arpit-Gulgulia** [comment]
Hi @williamdes, and @MauricioFauth, all the test cases failed with the same below error:-
`PhpMyAdmin\Tests\Config\PageSettingsTest::testShowGroupBrowse
Failed asserting that **<SOME_HTML>**`

Can you please suggest here? 
Thanks.

**Review by williamdes — COMMENTED** [review]
That definitely looks better without using the function empty

**Comment by williamdes** [comment]
Note: is better to rebase than to use merge master into the branch. Also squashing commits that are fixes of the first one also help

:)

**Comment by Arpit-Gulgulia** [comment]
> That definitely looks better without using the function empty

Yeah, but having hard luck passing these tests.

File: libraries/classes/Footer.php [hunk 1]
```diff
@@ -143,13 +145,11 @@ public function getSelfUrl(): string
         $params = [];
         $params['route'] = Common::getRequest()->getRoute();
 
-        if (isset($GLOBALS['db']) && strlen($GLOBALS['db']) > 0) {
-            $params['db'] = $GLOBALS['db'];
-        }
+	$db = DatabaseName::tryFromValue($GLOBALS['db']);
+	$params['db'] = $db !== null ? $db->getName() : '';
```
Comment (williamdes) [inline libraries/classes/Footer.php:149]: I am note sure we want to have an empty db or table name here
Maybe this is what is breaking tests

**Comment by williamdes** [comment]
Do you manage to run the test suite locally?

**Comment by williamdes** [comment]
> > Do you manage to run the test suite locally?
> 
> I'm doing changes on my Ec2 instance and it isn't configured to run the test suite.

Oh okay, if you have composer just use
```sh
composer run phpunit -- --no-coverage
```
After a one time `composer update`

File: libraries/classes/Footer.php (hunk 1 above)
Comment (Arpit-Gulgulia) [reply libraries/classes/Footer.php:149]: From the existing code lines, it seems like it is trying to get the DB name only. Maybe the indentation is the issue, not sure why it is not indented, it was indeed indented on my ec2 linux server.

**Comment by Arpit-Gulgulia** [comment]
I see in the logs that, the build is failing with the below errors:

7) PhpMyAdmin\Tests\GitTest::testExtractDataFormTextBody
Failed to assert that two arrays are identical.

3) PhpMyAdmin\Tests\Config\ServerConfigChecksTest::testBlowfishWithValidSecret
Failed asserting that an array does not have the key 'error'.

4) PhpMyAdmin\Tests\Config\ServerConfigChecksTest::testBlowfishWithInvalidSecret
Failed asserting that an array does not have the key 'error'.

Any help would be really appreciated.

**Comment by williamdes** [comment]
> 

The CI logs show that
```
2022-10-14T16:12:48.8037904Z There was 1 failure:
2022-10-14T16:12:48.8040759Z 
2022-10-14T16:12:48.8063848Z 1) PhpMyAdmin\Tests\Config\PageSettingsTest::testShowGroupBrowse
```
https://github.com/phpmyadmin/phpmyadmin/actions/runs/3251373711/jobs/5336272080#step:8:73

It seems to be the only test to fix

**Comment by Arpit-Gulgulia** [comment]
@williamdes, is this PR good to merge? Can you please suggest. Thanks

**Comment by Arpit-Gulgulia** [comment]
> @williamdes, is this PR good to merge? Can you please suggest. Thanks

@williamdes, did you get sometime to check on this? Please let me know if any changes requires.

**Comment by Arpit-Gulgulia** [comment]
@MauricioFauth Could you suggest here?

File: libraries/classes/Controllers/Normalization/AddNewPrimaryController.php [hunk 2]
```diff
@@ -25,17 +27,24 @@ public function __construct(ResponseRenderer $response, Template $template, Norm
     public function __invoke(ServerRequest $request): void
     {
         $num_fields = 1;
+
+        //Ref #17727 - DatabaseName and TableName should be used for $db and $table
```
Comment (williamdes) [inline libraries/classes/Controllers/Normalization/AddNewPrimaryController.php:31]: Can you remove this line and squash your commits?

Comment (Arpit-Gulgulia) [reply libraries/classes/Controllers/Normalization/AddNewPrimaryController.php:31]: Sure, Thanks

Comment (Arpit-Gulgulia) [reply libraries/classes/Controllers/Normalization/AddNewPrimaryController.php]: done.

**Comment by Arpit-Gulgulia** [comment]
@MauricioFauth, did you get some time to check on this PR?
Please review and suggest any changes required.
Thanks

**Review by MauricioFauth — APPROVED** [review]

**Comment by arpit-jain-sage** [comment]
Always good to see the changes merged into master😍.
Could you please also marked this as hacktoberfest-accepted. Thanks!

**Comment by Arpit-Gulgulia** [comment]
Always good to see your changes getting merged into main branch😍.
Could you please also mark this as hacktoberfest-accepted.
Thanks.

**Pull request merged**